client/state/posts/selectors.js

```diff
@@ -90,11 +97,17 @@ export const getSitePost = createSelector(
  * @return {?Array}         Posts for the post query
  */
 export function getSitePostsForQuery( state, siteId, query ) {
-	if ( ! state.posts.queries[ siteId ] ) {
+	const manager = state.posts.queries[ siteId ];
+	if ( ! manager ) {
+		return null;
+	}
+
+	const posts = manager.getItems( query );
+	if ( ! posts ) {
 		return null;
 	}
 
-	return state.posts.queries[ siteId ].getItems( query );
+	return posts.map( ( post ) => getNormalizedPost( state, post.global_ID ) );
```

Comment:
It depends what we want the source of truth to be. The posts tree is a bit problematic in that there appear to be two sources of truth, the `PostQueryManager` contents in `state.posts.queries` and the `state.posts.items` values. It's possible that `items` could include some items not found in `queries`, since it hooks into `POSTS_RECEIVE` (the `queries` key hooks on this too, but only updating `PostQueryManager` instances for sites that had already received query results). 

(As background, I noted this worry about duplication in a previous comment at https://github.com/Automattic/wp-calypso/pull/5135#issuecomment-221672980 ).

I'm still not sure what the right solution would be to deduplicate this, but it's a problem we'll want to solve before allowing queries to be persisted (so we're not wasting IndexedDB storage). I lean towards letting `PostQueryManager` (i.e. `state.posts.queries`) be the only place to find the full objects, either (a) removing the option to get a single post by global ID, (b) updating `getPost` to naively iterate over all items of all sites in `state.items.queries` to find the matching post, or (c) change `state.posts.items` to be a mapping of global ID to site ID / post ID path (this is my _most_ preferred option, but kinda goes against our desire for removing redundancy from reducers).

This is the direction I went for terms in #6224, removing `items` altogether. In that case, however, terms has the advantage of not being globally referenced.